### PR TITLE
Feature/Enhance the color widget

### PR DESCRIPTION
**Pull request by nathanNguyenQ1**
### Reasons for making this change

Currently, the color widget is not responsive enough for user interaction, it doesn't show the color code until the user clicks to open the palette color. The new UI allows users to look and feel the interaction.

UI Before:
image

UI After:
image

- User can view, edit, or paste color code directly without open palette color.
- User can open palette color to pick a color and the value will display on the text box

### Checklist

* [x] **I'm updating documentation**
  - [x] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [x] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [x] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [x] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/material-ui/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@rjsf/material-ui",
-  "version": "3.2.1",
+  "version": "3.2.2",
```
Comment (epicfaace) [inline packages/material-ui/package.json:3]: Can you undo these version bump changes

Comment (nathanNguyenQ1) [reply packages/material-ui/package.json]: Hi @epicfaace 
I updated the code to undo these version changes. Thank you,

File: lerna.json [hunk 2]
```diff
@@ -2,5 +2,5 @@
   "packages": [
     "packages/*"
   ],
-  "version": "3.2.1"
+  "version": "3.2.2"
```
Comment (epicfaace) [inline lerna.json:5]: Hmm, do you know why this version change is still showing up? Maybe you accidentally re-added the version changes?

Comment (nathanNguyenQ1) [reply lerna.json]: Hi @epicfaace
I updated the code for version changes. Thank you

**Comment by heath-freenome** [comment]
@nathanNguyenQ1 Can you rebase and deal with conflicts?

**Comment by nathanNguyenQ1** [comment]
Hi @epicfaace and @heath-freenome  I have rebased and fix conflict in merge request. Could you please review it. Thank you.

**Comment by heath-freenome** [comment]
@nathanNguyenQ1 If you revert all of the `package*.json` changes (when we release we'll do all that) and deal with the conversion of `ColorWidget` to typescript, I'm willing to add this to the v5 beta

File: packages/core/src/components/widgets/ColorWidget.js [hunk 3]
```diff
@@ -9,7 +9,43 @@ function ColorWidget(props) {
       widgets: { BaseInput },
```
Comment (heath-freenome) [inline packages/core/src/components/widgets/ColorWidget.js:9]: This is now called `BaseInputTemplate` and is obtained via a call like this:
```
  const { options, registry } = props;
  const BaseInputTemplate = getTemplate<"BaseInputTemplate", T, F>(
    "BaseInputTemplate",
    registry,
    options
  );
```

Comment (nathanNguyenQ1) [reply packages/core/src/components/widgets/ColorWidget.js]: Hi @heath-freenome 
I reverted all of the package*.json and updated the code for the new version. Thank you

**Comment by heath-freenome** [comment]
@nathanNguyenQ1 You'll have to update the snapshots for many of the theme tests. Also, Please update the `CHANGELOG.md` for the next, unpublished beta release

**Comment by nathanNguyenQ1** [comment]
> @nathanNguyenQ1 You'll have to update the snapshots for many of the theme tests. Also, Please update the `CHANGELOG.md` for the next, unpublished beta release

Hi @heath-freenome, I updated the snapshot and `CHANGELOG.md`.

File: CHANGELOG.md [hunk 4]
```diff
@@ -15,6 +15,11 @@ it according to semantic versioning. For example, if your PR adds a breaking cha
 should change the heading of the (upcoming) version to include a major version bump.
 
 -->
+# v5.0.0-beta.7
+
+## @rjsf/core
```
Comment (heath-freenome) [inline CHANGELOG.md:20]: Technically the next release is v5.0.0-beta.6, so if you move this down around line 26 that would be awesome!

**Review by heath-freenome — APPROVED** [review]

**Comment by heath-freenome** [comment]
@nathanNguyenQ1 I just tried this with the other themes and most of them do not look so good. I'm leaning towards reverting this PR until the other themes are fixed up as well. Is that something you are willing to do?

**Comment by heath-freenome** [comment]
@nathanNguyenQ1 After sleeping on it, I am definitely going to revert this. I can help you understand what needs to be done to get things working on the other themes if you'd like. Reach out on the [discord](https://discord.gg/fdZm2JZJrX) channel

**Pull request merged**